modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -99,10 +99,11 @@ def enum_tree(tree, share, subdir = '')
       list = tree.list(directory: subdir)
     rescue RubySMB::Error::UnexpectedStatusCode => e
       vprint_error("Error when trying to list tree contents in #{share[:name]}\\#{subdir} - #{e.status_code.name}")
+      return read, write, msg, []
```

Comment:
Should this not be `nil` vs `[]` given the line `return read, write, msg, nil if skip`? We should really be returning either `nil` or an empty array but not both from the same method otherwise checks are going to be crazy for validating this method.

Reply:
Yup, you're right sorry.
I was returning `[]` instead of `nil` due to broken logic that checked for empty or nil directories. This has been fixed in the latest commit.
Thanks!

Reply:
Resolved in ce3215e35b83ac9cb61b08d82b498d44354589ec